src/charm.py

```diff
@@ -74,19 +78,56 @@ def __init__(self, *args):
             ISTIO_CRDS_LABEL: self._get_crds_kubernetes_resource_manager,
             GATEWAY_API_CRDS_LABEL: self._get_gateway_apis_kubernetes_resource_manager,
         }
-
+        self._metrics_labels = ",".join(
+            [f"{key}={value}" for key, value in METRICS_LABELS.items()]
+        )
         self._lightkube_field_manager: str = self.app.name
+        # Configure Observability
+        self._scraping = MetricsEndpointProvider(
+            self,
+            relation_name="metrics-endpoint",
+            jobs=[{"static_configs": [{"targets": ["*:15090"]}]}],
+        )
 
         self.framework.observe(self.on.config_changed, self._reconcile)
         self.framework.observe(self.on.remove, self._remove)
```

Comment:
We should observe pebble_ready. It's unlikely to matter but if all of our initial hooks run before pebble is ready, we will never set up the layer.